### PR TITLE
black, flake8, and isort (part 2)

### PR DESCRIPTION
**Pull request by crusaderky**
- Configure flake8 and isort
- Resolve all flake8 errors
- Assorted code quality tweaks (embeds #664)
- Out of scope: CI integration and documentation (will be done in a successive PR)

### CODE REVIEW

**Comment by hgrecco** [comment]
What is the purpose of  `default_section=THIRDPARTY `?

**Comment by crusaderky** [comment]
@hgrecco  https://github.com/timothycrosley/isort/wiki/isort-Settings

> default_section: The default section to place imports in, if their section can not be automatically determined. FIRSTPARTY, THIRDPARTY, etc.

File: pint/__init__.py [hunk 1]
```diff
@@ -124,7 +125,9 @@ def test():
     "RedefinitionError",
     "UndefinedUnitError",
     "UnitStrippedWarning",
+    "formatter",
     "get_application_registry",
     "set_application_registry",
+    "pi_theorem",
```
Comment (crusaderky) [inline pint/__init__.py]: @hgrecco do we actually want to expose these?

**Comment by crusaderky** [comment]
@hgrecco this is ready for review and possibly merge. I can put the CI integration in a successive PR.

**Pull request merged**